### PR TITLE
Vfx/2019.3 staging

### PR DESCRIPTION
**Pull request by julienf-unity**
---
### Purpose of this PR
Bug backport for 7.1.7

List of fixes:
- Added ExcludeFromPreset on vfxparameter because presets don't work with parameters 
- Fix for exception when opening a graph with change sugraph operator slots
- Fix exception when compiling an asset with a turbulence block in absolute mode https://fogbugz.unity3d.com/f/cases/1203065/
- make property names clip
- Fix subgraph block into subgraph contex 
- Allow reattaching when edit button opens the visual effect graph and attach 
- Fixed GetCustomAttribute that was locked to Current https://fogbugz.unity3d.com/f/cases/1166686/
- Shader compilation now works when using view direction in shader graph https://fogbugz.unity3d.com/f/cases/1204027
- Fix for destroying selected component corrupt "Play Controls" window
- Fix for DockAreaHost exception 

---
**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/vfx%252F2019.3-staging

@PaulDemeulenaere @iTris666 Please review changes to make sure everything is good

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/CHANGELOG.md [hunk 1]
```diff
@@ -178,6 +178,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fixed an issue with Metal Shader Compiler and GTAO shader for metal
 - Fixed resources load issue while upgrading HDRP package.
 - Fix LOD fade mask by accounting for field of view
+<<<<<<< HEAD
```
Comment (PaulDemeulenaere) [inline com.unity.render-pipelines.high-definition/CHANGELOG.md]: Unexpected diff in changelog : com.unity.render-pipelines.high-definition/CHANGELOG.md (bad conflict resolution at https://github.com/Unity-Technologies/ScriptableRenderPipeline/commit/227d1ed70d58111e2ce4ada69d5e7e27cd20c5ca#diff-304ed94d014d1ee8fda0a407a849e9a2R181)

File: com.unity.visualeffectgraph/CHANGELOG.md [hunk 2]
```diff
@@ -38,13 +38,19 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Fix output settings correctly filtered dependeing on shader graph use or not
 - Fix some cases were normal/tangent were not passes as interpolants with shader graph
 - Make normals/tangents work in unlit output with shader graph
+- Sample Scene Color & Scene Depth from Shader Graph Integration using High Definition and Universal Render Pipeline
```
Comment (PaulDemeulenaere) [inline com.unity.visualeffectgraph/CHANGELOG.md]: Bad conflict resolution at https://github.com/Unity-Technologies/ScriptableRenderPipeline/commit/227d1ed70d58111e2ce4ada69d5e7e27cd20c5ca#diff-462fa9b3b0a6774039651b8688338aeeR49
This entry was already present in changelog.

File: com.unity.visualeffectgraph/Shaders/ParticleLinesSW/PassDepth.template [hunk 3]
```diff
@@ -8,7 +8,7 @@ Pass
 	
 	HLSLPROGRAM
 	#define VFX_PASSDEPTH VFX_PASSDEPTH_ACTUAL
-	${VFXInclude("Shaders/ParticleLinesSW/PassDepthOrMV.template")}
 	${VFXPassDepthDefine}
+	${VFXInclude("Shaders/ParticleLinesSW/PassDepthOrMV.template")}
```
Comment (PaulDemeulenaere) [inline com.unity.visualeffectgraph/Shaders/ParticleLinesSW/PassDepth.template]: FYI, these template change are expected, I did a bad merge, fixed with this merge request : https://github.cds.internal.unity3d.com/unity/vfx-graph/pull/204/files (there is no impact on final generated shader)

**Comment by PaulDemeulenaere** [comment]
Failure in SubgraphContextWithSubgraphBlock image reference in `VFXMain on Win_DX11_Standalone` (@iTris666) :
Expected :
image
Actual :
image
Diff :
image

0.000633770949f image difference, treshold is 0.0005.

VFX graphic test should not display the default sky background by the way.

File: TestProjects/VisualEffectGraph/Assets/ReferenceImages/Linear/WindowsPlayer/Direct3D11/SubgraphContextWithSubgraphBlock.png.meta [hunk 4]
```diff
@@ -0,0 +1,92 @@
+fileFormatVersion: 2
```
Comment (PaulDemeulenaere) [inline TestProjects/VisualEffectGraph/Assets/ReferenceImages/Linear/WindowsPlayer/Direct3D11/SubgraphContextWithSubgraphBlock.png.meta]: Update the screenshot with standalone result (or replace by a black background)

File: com.unity.render-pipelines.high-definition/CHANGELOG.md (hunk 1 above)
Comment (julienf-unity) [reply com.unity.render-pipelines.high-definition/CHANGELOG.md]: Fixes

**Comment by julienf-unity** [comment]
Still a gfx test to be fixed by @iTris666

File: TestProjects/VisualEffectGraph/Assets/ReferenceImages/Linear/WindowsPlayer/Direct3D11/SubgraphContextWithSubgraphBlock.png.meta (hunk 4 above)
Comment (julienf-unity) [reply TestProjects/VisualEffectGraph/Assets/ReferenceImages/Linear/WindowsPlayer/Direct3D11/SubgraphContextWithSubgraphBlock.png.meta]: Yes either remove the sky or slightly increase the threshold for this particular test

Comment (PaulDemeulenaere) [reply TestProjects/VisualEffectGraph/Assets/ReferenceImages/Linear/WindowsPlayer/Direct3D11/SubgraphContextWithSubgraphBlock.png.meta]: I'm setting manually the sky to black & update screenshot, I will do the same on `vfx/staging`

**Comment by PaulDemeulenaere** [comment]
I'm waiting for yamato before validate this changes.

**Comment by VladNeykov** [comment]
Test doc: https://docs.google.com/document/d/1-ybkklf2tR4BAbZ78CiMarWlM6g3Do4fXLRxg4l38cE/edit#

Mostly looking good, a couple manual tests failed, please take a look.

**Pull request merged**